app/src/main/kotlin/com/softteco/template/ui/components/PrimaryButton.kt

```diff
@@ -0,0 +1,61 @@
+package com.softteco.template.ui.components
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.softteco.template.ui.theme.Dimens
+
+@Composable
+fun PrimaryButton(
+    buttonText: String,
+    loading: Boolean,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+) {
+    val isButtonEnabled = remember { mutableStateOf(true) }
+    Box(
+        modifier = modifier.padding(
+            Dimens.PaddingLarge
+        )
+    ) {
+        Button(
+            enabled = isButtonEnabled.value,
```

Comment:
Let's do not change `enabled` parameter, it changes button color. 
We can disable clicking by:
```
Button(
    onClick = { if (!loading) onClick() },
```

Reply:
and then we don't need `val isButtonEnabled = remember { mutableStateOf(true) }`

Reply:
But if we don't set `enabled` parameter, while loading if we click the button, it is clickable. And the color of button also changes when click

Reply:
Changing color during clicking - it's expected, it's ok.
But when we change `enabled` parameter, the button changes color while the loader is loading. Usually Or button has a loader or button set as disabled. It shouldn't have both. 
Let's just disable clicking by adding 
`onClick = { if (!loading) onClick() },`